File: Java/src/main/java/Math/ArmstrongNumber/ArmstrongNumber.java
package Math.ArmstrongNumber;

public class ArmstrongNumber {

    public static void main(String [] args) {

        int [] inputList = {153, 123};
        for (int i : inputList)
            System.out.println("Input: " + i + "    isArmStrong: " + isArmstrong(i));
    }

    public static boolean isArmstrong(int N) {

        String [] arr = String.valueOf(N).split("");
        int length = arr.length;
        int total = 0;

        for (String strDigit : arr) {
            int digit = Integer.valueOf(strDigit);
            total += Math.pow(digit, length);
        }

        System.out.println("Total: " + total);

        if (total != N)
            return false;

        return true;
    }

}

File: Python/project/src/Math/ArmStrongNumber/ArmStrongNumber.py


class Solution(object):
    def isArmstrong(self, num):

        # Step1 - Find the num of digits in num
        numDigits = str(num).__len__()

        # Step2 - For every digit in num, we calculate the total armstrong number
        total = 0
        for strDigit in str(num):
            total += pow(int(strDigit), numDigits)
        print("Total: ", total)

        # Step3 - Return if total is equal to the input 'num'
        return total == num



def main():
    N = 153
    solution = Solution()
    output = solution.isArmstrong(N)
    print("Is ArmStrong Number: " + str(output))


if __name__ == '__main__':
    main()



'''
Notes
- Quite Easy

Solution
1. Find num of digits
2. Traverse every digit in num and calculaet the total
3. Compare the total with the input 'num'
'''